### PR TITLE
enable GUI for 3rd party simulator - DC 

### PR DESCRIPTION
**Pull request by ramyam-xilinx**
Reviewer : venkatp

### CODE REVIEW

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.h [hunk 1]
```diff
@@ -263,6 +263,10 @@ using addr_type = uint64_t;
 
       std::string getSimulatorType(const std::string& binaryDirectory);
       void createPreSimScript(const std::string& wcfgFilePath, std::string& preSimScriptPath);
+      std::string loadFileContentsToString(const std::string& path);
+      void writeStringIntoFile(const std::string& path, const std::string& content);
+      std::string modifyContent(const std::string& simulatorName, std::string& content);
+      void writeNewSimulateScript (const std::string& simPath, const std::string& simulatorName );
```
Comment (keryell) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.h:269]:
```suggestion
      void writeNewSimulateScript (const std::string& simPath, const std::string& simulatorName);
```

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx [hunk 2]
```diff
@@ -143,6 +143,49 @@ namespace xclhwemhal2 {
     }
 
   }
+  
+  std::string HwEmShim::loadFileContentsToString(const std::string& path) 
+  {
+    std::ifstream file(path);
+    return std::string((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
```
Comment (keryell) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx:150]: Useless `()`.
Also you can probably simplify this to
```suggestion
    return { std::istreambuf_iterator<char>(file), std::istreambuf_iterator<char>() };
```

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx [hunk 3]
```diff
@@ -629,6 +672,7 @@ namespace xclhwemhal2 {
           mLogStream << __func__ << " UNZIP of sim bin ended and permissions operation is complete" << std::endl;
 
         simulatorType = getSimulatorType(binaryDirectory);
+        std::transform(simulatorType.begin(), simulatorType.end(), simulatorType.begin(), ::tolower);
```
Comment (keryell) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx:675]: Do not use random C code in C++
```suggestion
        std::transform(simulatorType.begin(), simulatorType.end(), simulatorType.begin(), std::tolower);
```
and add `#include <cctype>` (while sorting the header includes...).

**Review by keryell — CHANGES_REQUESTED** [review]
That looks good but I can see a few simplifications

**Comment by xilinxgitops** [comment]
retest this please

**Pull request merged**